build.assets/Dockerfile

```diff
@@ -1,18 +1,17 @@
-# This Dockerfile makes the "build box": the container used to build official
-# releases of Teleport and its documentation.
+# This Dockerfile makes the "build box" the container used to:
+# * build Teleport Connect
+# * run test and linters in CI
+# * building other Docker images
+#
+# For Teleport releases we're using CentOS 7 box to keep the binaries compatible
+# with older Linux distributions (glibc 2.17+).
 #
 # Check the README to learn how to safely introduce changes to Dockerfiles.
 
-# Use Ubuntu 18.04 as base to get an older glibc version.
-# Using a newer base image will build against a newer glibc, which creates a
-# runtime requirement for the host to have newer glibc too. For example,
-# teleport built on any newer Ubuntu version will not run on Centos 7 because
-# of this.
-
 ## LIBFIDO2 ###################################################################
 
 # Build libfido2 separately for isolation, speed and flexibility.
-FROM buildpack-deps:18.04 AS libfido2
+FROM buildpack-deps:20.04 AS libfido2
```

Comment:
If we build Connect on 22.04 then it will require glib from 22.04, so it won’t work on 20.04.
A native module that we rebuild causes a dependency on glib.

Reply:
Should we built connect on centos7, then?

There's also the possibility of bootstrapping glibc ourselves - it takes some work, but frees us from having to use old distros to build. I think @jakule and myself talked about this before.

(In any case this comment ain't a blocker.)

Reply:
IMO we should avoid building more stuff on CentOS 7. Node.js doesn't provide builds for CentOS 7 (or even Ubuntu 18.04 at this point). If we want to build everything on CentOS 7 we would need to build Node.js from the source, but even if this is possible Node doesn't provide official support for such an old version of Glibc. They are also not testing it on some very old platforms, which case cause some more problems in the future.
The main question at this point is how many people are even using Connect on Ubuntu older than 20.04 (and other distros with older glibc). I'd assume that CentOS 7 is not very popular for Linux desktop distributions, and we should be fine with Ubuntu 20.04+.

The one thing that I'd advocate for is to split the main buildbox and Connect container. By doing so, we will be able to use `ubuntu:latest` in our CI and keep whatever OS we want for Connect build. This should also speed up the Connect build itself, as Connect doesn't need 90% of the `Buildbox` dependencies (Go, Rust, linters, etc).

Reply:
Also keep in mind that the current version of Electron embeds Node v18 that requires glibc >= 2.28 (so Connect won't even start on Centos7, even if we build it on it). 
Ubuntu 20.04 uses glibc 2.31. Theoretically, we are losing potential users with OSes that use glibc 2.28 - 2.30, but I don't think it is a big deal?

Reply:
CentOS 8 Stream and Debian 10 are using 2.28 (random website that seems to have correct data https://repology.org/project/glibc/versions). Both are EOL the next year. Again, it's hard to estimate the market usage, but it looks like these users will be affected. If this is an issue we could split the image (as I mentioned above) and switch the base image to Debian 10 (should be simple as Ubuntu is based on Debian).
CC @zmb3

Reply:
> The one thing that I'd advocate for is to split the main buildbox and Connect container

+1